include/libopencm3/cm3/common.h

```diff
@@ -59,6 +61,29 @@
 #define BBIO_PERIPH(addr, bit) \
 	MMIO32((((uint32_t)addr) & 0x0FFFFF) * 32 + 0x42000000 + (bit) * 4)
 
+#else
+
+/* Basic Assembler support */
+
+/* Skip all C/C++ declarations */
+#define BEGIN_DECLS .if 0
+#define END_DECLS .endif
+
+/* All addresses are just numbers */
+#define MMIO8(addr) (addr)
+#define MMIO16(addr) (addr)
+#define MMIO32(addr) (addr)
+#define MMIO64(addr) (addr)
+
+/* Generic bit-band I/O accessor functions */
+#define BBIO_SRAM(addr, bit) \
+	(((addr) & 0x0FFFFF) * 32 + 0x22000000 + (bit) * 4)
+
+#define BBIO_PERIPH(addr, bit) \
+	(((addr) & 0x0FFFFF) * 32 + 0x42000000 + (bit) * 4)
```

Comment:
completely out of scope of this commit

Reply:
Sorry, but why? The BBIO_* C/C++ macros are just above and they do the same. But I can remove them from the commit if that helps.